Upstream-Status:Inappropriate [debian patch]

From: Niko Tyni <ntyni@debian.org>
Description: Downgrade the optimization of sv.c on arm due to a gcc-4.4 bug
Bug-Debian: http://bugs.debian.org/580334

Regression from gcc-4.3, not reported yet.

 perl -e '"-2" =~ /(.+)/; @foo=(); push @foo, $_ for $1..undef; print @foo ? "ok\n" : "not ok\n"'


---
 cflags.SH |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/cflags.SH b/cflags.SH
index a5d71b9..6914ba6 100755
--- a/cflags.SH
+++ b/cflags.SH
@@ -312,7 +312,11 @@ for file do
     regexec) ;;
     run) ;;
     scope) ;;
-    sv) ;;
+    sv)
+	case $archname in
+	arm-*|armeb-*)
+	    optimize=-O0;;
+	esac;;
     taint) ;;
     toke) ;;
     universal) ;;
-- 
tg: (a508b62..) debian/arm_optim (depends on: upstream)
